repo/maintenance/blob_retain.go

```diff
@@ -19,72 +19,80 @@ type ExtendBlobRetentionTimeOptions struct {
 
 // ExtendBlobRetentionTime extends the retention time of all relevant blobs managed by storage engine with Object Locking enabled.
 func ExtendBlobRetentionTime(ctx context.Context, rep repo.DirectRepositoryWriter, opt ExtendBlobRetentionTimeOptions, safety SafetyParameters) (int, error) {
-	blobCfg := rep.BlobCfg()
-	if !blobCfg.IsRetentionEnabled() || !blobCfg.RetentionExtend {
-		// Blob retention is disabled
-		return 0, nil
-	}
+	const extendQueueSize = 100
+
+	var (
+		cnt, toExtend uint64
+		eg            errgroup.Group
+		prefixes      []blob.ID
+	)
 
 	if opt.Parallel == 0 {
 		opt.Parallel = 16
 	}
 
-	const extendQueueSize = 100
+	blobCfg := rep.BlobCfg()
+	if !blobCfg.IsRetentionEnabled() {
+		// Blob retention is disabled
+		log(ctx).Info("Object lock retention is disabled.")
+		return 0, nil
+	}
 
 	extend := make(chan blob.Metadata, extendQueueSize)
-
-	var cnt, toExtend uint64
-
-	var eg errgroup.Group
-
-	var prefixes []blob.ID
-
 	extendOpts := blob.ExtendOptions{
 		RetentionMode:   blobCfg.RetentionMode,
 		RetentionPeriod: blobCfg.RetentionPeriod,
 	}
 
 	if !opt.DryRun {
-		// start goroutines to delete blobs as they come.
+		// start goroutines to extend blob retention as they come.
 		for i := 0; i < opt.Parallel; i++ {
 			eg.Go(func() error {
+				var failed []blob.ID
 				for bm := range extend {
 					if err := rep.BlobStorage().ExtendBlobRetention(ctx, bm.BlobID, extendOpts); err != nil {
-						return errors.Wrapf(err, "unable to extend blob %q", bm.BlobID)
+						failed = append(failed, bm.BlobID)
```

Comment:
nit: i would just log the error as things happend and increment a counter, no need to buffer all those blob IDs.